examples/model_selection/plot_grid_search_digits.py

```diff
@@ -13,62 +13,149 @@
+# select the best candidate from the `cv_results_` returns from the
+# :class:`~sklearn.model_selection.GridSearchCV`. Once the candidate selected,
+# it will be automatically refitted by the
+# :class:`~sklearn.model_selection.GridSearchCV` instance. Here, the strategy
+# is to short list the model which are the best in terms of precision and
+# recall. From the selected models, we finally select the fastest model at
+# predicting.
+#
+# Note that these choices are completely arbitrary.
+import pandas as pd
+
+
+def refit_strategy(cv_results):
+    """Define the strategy to select the best estimator.
+
+    The strategy defines here is to select the best ranking model in terms of
+    precision and recall. Once these models are selected, we can select the
+    fastest model to predict.
+
+    Parameters
+    ----------
+    cv_results : dict of numpy (masked) ndarrays
+        CV results as returned by the `GridSearchCV`.
+
+    Returns
+    -------
+    best_index : int
+        The index of the best estimator as it appears in `cv_results`.
+    """
+    # print the info about the grid-search for the different scores
+    for score in scores:
+        print("\nGrid scores on development set using {score}:\n")
+        for mean, std, params in zip(
+            cv_results[f"mean_test_{score}"],
+            cv_results[f"std_test_{score}"],
+            cv_results["params"],
+        ):
+            print(f"{mean:0.3f} (+/-{std * 2:0.03f}) for {params}")
+
+    cv_results_ = pd.DataFrame(cv_results)
+    cv_results_ = cv_results_[
+        [
+            "mean_score_time",
+            "mean_test_recall_macro",
+            "mean_test_precision_macro",
+            "rank_test_recall_macro",
+            "rank_test_precision_macro",
+        ]
+    ]
+
+    # Select the most performant models in terms of precision/recall
+    mask_best_recall = cv_results_["rank_test_recall_macro"] == 1
+    mask_best_precision = cv_results_["rank_test_precision_macro"] == 1
+    mask_combined = mask_best_recall & mask_best_precision
+    print(
+        "The following models are the most performance in terms of "
+        f"recall/precision:\n\n {cv_results_[mask_combined]}"
+    )
+
+    # From the best candidates, select the fastest model to predict
+    best_index = cv_results_.loc[mask_combined, "mean_score_time"].idxmin()
+
+    print(
+        "\nWe selected the model with the following parameters which maximize"
+        " the performance metric and minimize the scoring time:\n\n"
+        f"{cv_results_.loc[best_index]}"
+    )
+
+    return cv_results_.index[best_index]
+
+
+# %%
+# Once we defined our strategy to select the best model, we can create the
+# grid-search instance. Subsequently, we can check the best parameters found.
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+
+clf = GridSearchCV(SVC(), tuned_parameters, scoring=scores, refit=refit_strategy)
+
+print("# Tuning hyper-parameters \n")
```

Comment:
We could make a section here to delimitate the definition of the framework and the actual call to fit as well as the reporting of the results.